app/static/js/main.js

```diff
@@ -276,29 +315,32 @@ function addPolygonToList(polygonID, computation) {
     $("#show-hide-" + polygonID).on("click", function(e) {
         var polygon = polygons.collection[polygonID];
         showHidePolygonButton(this, polygon);
+        isHidden = !isHidden;
     })
     $("#delete-" + polygonID).on("click", function(e) {
         var polygonID = $(this).parent().attr("id");
         var polygon = polygons.collection[polygonID];
         deletePolygonButton(this, polygon);
     })
     $("#" + polygonID).on("click", function(e) {
-       clearPolygonListBorders();
        var polygon = polygons.collection[polygonID];
-       polygons.setSelection(polygon);
-       createPolygonListBorder(polygonID);
+       if (!polygons.isInSCollec(polygon) && !isHidden) {
+           polygons.multipleSelection(polygon);
+           createPolygonListBorder(polygonID);
+       } else {
+           polygons.mutipleClearSelection(polygon);
+           clearPolygonListBorders(polygonID);
+       }
+       managePolygon(this.id,"selected");
    })
 }
 
-function createPolygonListBorder(polygonID){
-    clearPolygonListBorders();
+function createPolygonListBorder(polygonID) {
     document.getElementById(polygonID).style.border = "2px solid black";
 }
 
-function clearPolygonListBorders() {
-    for(var polygonID in polygons.collection) {
-        document.getElementById(polygonID).style.border = "none";
-    }
+function clearPolygonListBorders(polygonID) {
+    document.getElementById(polygonID).style.border = "none";
```

Comment:
This is *fine* however you're using JavaScript DoM to modify the HTML, it's just that we've been using jQuery to modify the HTML. They both do the same thing but in my opinion it doesn't flow with what we currently have